### PR TITLE
Update resources for v4 migration

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

- Group all materials related to the topic under "Adobe Mobile Library (v4) end-of-support" - renamed to avoid name conflicts with the ios 4.x sdks
- Revisited the v4 faq page, moved non faq content under overview
- Updated sample code to use latest version of the SDKs
- Some refactoring under tabs

## Screenshots (if appropriate):
See latest summary menu: 
Screenshot 2023-08-02 at 9 07 44 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
+<Variant platform="android" task="config" repeat="4"/>
```
Comment (emdobrin) [inline src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md:1]: this was moved from analytics-comparison.md and upgraded to latest SDKs

File: src/pages/documentation/resources/v4-end-of-life-faq.md [hunk 2]
```diff
@@ -1,53 +0,0 @@
-# Version 4 Mobile SDKs end-of-support FAQ
```
Comment (emdobrin) [inline src/pages/documentation/resources/v4-end-of-life-faq.md:1]: split into faq + overview pages

File: src/pages/documentation/resources/upgrade-platform-sdks/v4-faq.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+Spurred by customer feedback for more control, performance, and functionality over the years; and to support future product enhancements, we released the re-architected [Adobe Experience Platform Mobile SDKs](../../../index.md) in September 2018. The new Experience Platform Mobile SDKs replaces the version 4 SDKs and provide support for new features such as flexible configuration through the Data Collection UI, modular Adobe and partner extensions.
+
+Solutions such as [Places Service](https://experienceleague.adobe.com/docs/places/using/home.html) and [Adobe Experience Platform Assurance](https://experienceleague.adobe.com/docs/experience-platform/assurance/home.html) are exclusively available on Experience Platform Mobile SDKs.
+
+## Where do I learn more about upgrading/migrating to the Experience Platform Mobile SDKs?
+
+To upgrade to the Experience Platform SDKs, see [Upgrade to the Experience Platform SDKs](..).
```
Comment (calebk1) [inline src/pages/documentation/resources/upgrade-platform-sdks/v4-faq.md:13]: Is this link correct?

**Review by calebk1 — APPROVED** [review]
LGTM!

Just a few minor changes.

File: src/pages/documentation/resources/upgrade-platform-sdks/v4-faq.md (hunk 3 above)
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/v4-faq.md:13]: this was the only way I was able to point to the local index.md, seems to work on localhost

File: src/pages/documentation/resources/upgrade-platform-sdks/index.md [hunk 4]
```diff
@@ -1,23 +1,45 @@
...
 
-This section contains a checklist that you can use as you plan your upgrade to the Experience Platform SDKs.
+This section contains a checklist that you can use as you plan your upgrade from Adobe Mobile Library (v4) to the Experience Platform SDKs.
 
-<InlineAlert variant="info" slots="text"/>
+## Update on end of support for version 4 Mobile SDKs
+
+<InlineAlert variant="warning" slots="text"/>
+
+Support for the Mobile [version 4 SDKs](https://github.com/Adobe-Marketing-Cloud/mobile-services) has ended as of **August 31, 2021** ~~September 30, 2020~~.
```
[the remark this answers is not in the document]
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/index.md:9]: I guess we can keep it as is

File: src/pages/documentation/resources/upgrade-platform-sdks/api-changelog.md [hunk 5]
```diff
@@ -4,119 +4,119 @@ This page details SDK API changes between the Experience Platform SDKs and 4x SD
...
-| trackPushMessageClickThrough:userInfo ([iOS](https://experienceleague.adobe.com/docs/mobile-services/ios/messaging-ios/push-messaging/push-messaging.html) \| [Android](https://experienceleague.adobe.com/docs/mobile-services/android/messaging-android/push-messaging/push-messaging.html)) | Support modified. See [guide](../faq.md#how-can-i-track-user-engagement-of-push-notifications-using-the-experience-platform-mobile-sdk) |
-| Tracking App Crash ([iOS](https://experienceleague.adobe.com/docs/mobile-services/ios/analytics-ios/crashes.html) \| [Android](https://experienceleague.adobe.com/docs/mobile-services/android/analytics-android/crashes.html)) | Deprecated |
+| trackActionFromBackground | Deprecated |
+| trackLocation:data: | Deprecated |
+| trackBeacon:Data: | Support modified. See [guide](../../adobe-analytics/track-beacon.md) |
+| trackingClearCurrentBeacon | Deprecated |
+| registerAdobeDataCallback: | Deprecated |
+| lifetimeValue | Deprecated |
+| trackLifetimeValueIncrease:data: | |
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/api-changelog.md:34]: Why is this not marked as `Deprecated`?

File: src/pages/documentation/resources/upgrade-platform-sdks/comparison.md [hunk 6]
```diff
@@ -35,14 +35,14 @@ The following tables provide information about the differences between the new E
...
-| Marketing/Acquisition Links | Yes | Yes - [Mobile Services](https://developer.adobe.com/client-sdks/previous-versions/documentation/adobe-analytics-mobile-services/) extension |
-| Geo location and beacon tracking | Yes | Yes - Use [Places extension](../places/index.md) |
-| Geo points-of-interest management | Yes | Yes - Use [Places extension](../places/index.md) |
+| Postbacks - Get/POST URL requests | Yes |	Yes - [Signals](../../mobile-core/signal/index.md) extension |
+| Postbacks - PII Get/POST URL requests | Yes | Yes - [Signals](../../mobile-core/signal/index.md) extension |
+| Postbacks - Open app deeplink | Yes | Yes - [Signals](../../mobile-core/signal/index.md) extension |
+| Push Messaging | Yes | Yes - [Mobile Services](https://developer.adobe.com/client-sdks/previous-versions/documentation/adobe-analytics-mobile-services/) extension (deprecated) |
+| In-app Messaging | Yes | Yes - [Mobile Services](https://developer.adobe.com/client-sdks/previous-versions/documentation/adobe-analytics-mobile-services/) extension (deprecated) |
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:42]: Why are we linking deprecated Analytics Mobile Services Push & InApp here? For latest, I think we should just mention that the functionality is supported with ACC (push only), ACS or AJO extensions.

File: src/pages/documentation/resources/upgrade-platform-sdks/comparison.md [hunk 7]
```diff
@@ -35,14 +35,14 @@ The following tables provide information about the differences between the new E
...
+| Marketing/Acquisition Links | Yes | Yes - [Mobile Services](https://developer.adobe.com/client-sdks/previous-versions/documentation/adobe-analytics-mobile-services/) extension (deprecated)|
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:43]: Conflicted between saying no and linking deprecated Analytics Mobile Services, prefer former though.

File: src/pages/documentation/resources/upgrade-platform-sdks/comparison.md [hunk 8]
```diff
@@ -35,14 +35,14 @@ The following tables provide information about the differences between the new E
...
+| Geo location and beacon tracking | Yes | Yes - Use [Places extension](../../places/index.md) |
+| Geo points-of-interest management | Yes | Yes - Use [Places extension](../../places/index.md) |
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:45]: Yes - ~Use~ [Places extension](../../places/index.md)
 Let's keep it consistent.

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+<Variant platform="ios" task="api-changes-aep" repeat="8"/>
+
+The Mobile SDKs have moved the `trackAction` and `trackState` APIs to the MobileCore extension. In addition, the context data dictionary has been changed from `<NSString, NSObject>` to `<NSString, NSString>`. The syntax is:
+
+```objectivec
++ (void) trackAction: (nullable NSString*) action data: (nullable NSDictionary<NSString*, NSString*>*) data;
+```
+
+```objectivec
++ (void) trackState: (nullable NSString*) action data: (nullable NSDictionary<NSString*, NSString*>*) data;
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md:174]: Let us be consistent and provide the Swift syntax/example and objective C syntax/ example (and in that order).

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+// syntax
+public static void trackAction(final String action, final Map<String, Object> contextData)
+
+// usage
+Analytics.trackAction("linkClicked", new HashMap<String, Object>() {{
+  put("url", "https://www.adobe.com");
+}});
+```
+
+<Variant platform="ios" task="api-changes-v4" repeat="4"/>
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md:117]: I'm wondering why we need v4 syntax/ examples. We should only provide AEP syntax/ examples for the corresponding APIs, along with specifying any changes to parameter types (like you've already done).

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/lifecycle.md [hunk 11]
```diff
@@ -18,48 +18,55 @@ public class MyApp extends Application {​
...
-In Swift, importing `ACPCore` also imports the necessary Lifecycle APIs:
-
 ```swift
-import ACPCore
+import AEPCore
+import AEPLifecycle
 ```
 
-Register the Lifecycle extension with the SDK Core by adding the following in your app's `application:didFinishLaunchingWithOptions:` delegate method:
+Register the Lifecycle extension along with the other extensions you use with the Mobile Core by adding the following in your app's `application:didFinishLaunchingWithOptions:` delegate method:
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/tabs/lifecycle.md:40]: nit: For Swift, use `application(_:didFinishLaunchingWithOptions:)`

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/lifecycle.md [hunk 12]
```diff
@@ -87,38 +94,39 @@ Setting the application is only necessary on activities that are entry points fo
 
 #### Swift
 
-Start Lifecycle data collection by calling `lifecycleStart:` from the callback of the `ACPCore::start:` method in your app's `application:didFinishLaunchingWithOptions:` delegate method.
+Start Lifecycle data collection by calling `lifecycleStart` from the registration completion handler in your app's `application:didFinishLaunchingWithOptions:` delegate method.
 
 If your iOS application supports background capabilities, your `application:didFinishLaunchingWithOptions:` method might be called when iOS launches your app in the background. If you do not want background launches to count towards your lifecycle metrics, `lifecycleStart:` should only be called when the application state is not equal to `UIApplicationStateBackground`.
```
Comment (swarna04) [inline src/pages/documentation/resources/upgrade-platform-sdks/tabs/lifecycle.md:99]: nit: For Swift, use
```
lifecycleStart(additionalContextData:)
application(_:didFinishLaunchingWithOptions:)
UIApplication.State.background
```

**Review by swarna04 — CHANGES_REQUESTED** [review]

File: src/pages/documentation/resources/upgrade-platform-sdks/comparison.md (hunk 7 above)
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:43]: I wasn't planning for any content changes in this file, but I agree we should revisit all AMS sections and faqs. I can create a task for that

Comment (swarna04) [reply src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:43]: ok, sounds good. Feel free to merge, create tasks for anything you might want to incorporate.

**Review by swarna04 — APPROVED** [review]

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md (hunk 10 above)
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md:117]: this is not new content, just moved from analytics-comparison.md + upgrade to AEP samples

File: src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md (hunk 9 above)
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/tabs/analytics.md:174]: added the syntax from mobilecore api-reference

File: src/pages/documentation/resources/upgrade-platform-sdks/comparison.md (hunk 6 above)
Comment (emdobrin) [reply src/pages/documentation/resources/upgrade-platform-sdks/comparison.md:42]: same here, created a dedicated task for this https://github.com/AdobeDocs/aep-mobile-sdkdocs/issues/384

**Pull request merged**